sdks/csharp/AgonesSdkClient.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Newtonsoft.Json;
+using System.Collections;
+using System.Text;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace Agones.SDK
+{
+    /// <summary>
+    /// JSON payload object
+    /// </summary>
+    internal class KeyValueMessage
+    {
+        public string key;
+        public string value;
+    }
+
+    /// <summary>
+    /// Agones SDK Unity Client
+    /// </summary>
+    public class AgonesSdkClient : MonoBehaviour

```

Comment:
```suggestion
    public class Client : MonoBehaviour
```
It's it namespace Agones.sdk - having it be called `AgonesSdkClient` seems redundant to me. WDYT?

Reply:
Only reason I named it that is customary to have MonoBehaviours have a descriptive name so when you look at them in the editor, it's obvious what it is.  That said, I can go either way.

Reply:
Ah, that makes perfect sense - got you!